python-frontend/src/main/java/org/sonar/plugins/python/api/PythonVisitorContext.java

```diff
@@ -49,7 +49,8 @@ public PythonVisitorContext(FileInput rootTree, PythonFile pythonFile, @Nullable
     symbolTableBuilder.visitFileInput(rootTree);
     SymbolTableBuilderV2 symbolTableBuilderV2 = new SymbolTableBuilderV2();
     symbolTableBuilderV2.visitFileInput(rootTree);
-    rootTree.accept(new TypeInferenceV2(new ProjectLevelTypeTable(ProjectLevelSymbolTable.empty()), pythonFile));
+    var symbolTable = symbolTableBuilderV2.build();
```

Comment:
I think there is something weird in this change about whether `build` or `visitFileInput` has to be called to trigger the visit.
Here we are calling `build` without setting the `FileInput` associated to the `SymbolTableBuilderV2` which should do nothing, and only works because `symbolTableBuilderV2.visitFileInput(rootTree); ` was called explicitly.

The duplication between `withFileInput` and the public `visitFileInput` doesn't help here I believe.

Would it make sense to instead have `FileInput` as a constructor parameter for `SymbolTableBuilder` then have `build` as its unique public method which calls the visit and then returns the `SymbolTable`?